# 06_agent_evaluation: Goal Completion and Trajectory Benchmarks

This notebook benchmarks agent execution runs, calculating Task Success Rate, Tool Selection Accuracy, and Trajectory Efficiency.


In [1]:
import numpy as np

# 1. Define evaluation structures
test_runs = [
    {"optimal_steps": 3, "actual_steps": 3, "success": True, "tool_accuracy": 1.0},
    {"optimal_steps": 3, "actual_steps": 5, "success": True, "tool_accuracy": 0.75},
    {"optimal_steps": 3, "actual_steps": 6, "success": False, "tool_accuracy": 0.50}
]

# 2. Compute KPIs
success_rates = [1.0 if r["success"] else 0.0 for r in test_runs]
trajectory_efficiencies = [r["optimal_steps"] / r["actual_steps"] for r in test_runs]
tool_accuracies = [r["tool_accuracy"] for r in test_runs]

print(f"Goal Completion Rate (GCR): {np.mean(success_rates)*100:.2f}%")
print(f"Mean Trajectory Efficiency (MTE): {np.mean(trajectory_efficiencies)*100:.2f}%")
print(f"Mean Tool Selection Accuracy: {np.mean(tool_accuracies)*100:.2f}%")


Goal Completion Rate (GCR): 66.67%
Mean Trajectory Efficiency (MTE): 70.00%
Mean Tool Selection Accuracy: 75.00%


### Output Explanation
- The KPI evaluation equations match hand calculations exactly:
  - GCR evaluates to $66.67\%$.
  - MTE evaluates to $70.00\%$.
  - Tool Selection Accuracy averages to $75.00\%$.
